### DISCUSSION TITLE
C# Unity Interaction

### DISCUSSION THREAD

**Discussion by EgoInc** [WalletConnect/walletconnect-monorepo#2996] [Developer Support]
Hello, all last last questiona about Unity have not received an answer. What's wrong, you shutdown v1 before stabilization of the second version...

 I also have problems with WCv2 connection

I use you instruction [here](https://github.com/WalletConnect/WalletConnectSharp) 

So, when I get to this step
```
Task<SessionData> sessionConnectTask = connectData.Approval;
SessionData sessionData = await sessionConnectTask;
```

And while this code is waiting for execution, WebsocketConnection throws TimeOut error when trying to create, in fact, a socket

**Comment by mustafadervisglu** [comment]
did you solve this error? @EgoInc

**Reply by EgoInc** [reply]
No
